docs/advanced/debugging.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Debugging
+order: 6
+---
+
+# Debugging
+
+We've tried to make debugging in `urql` as seamless as possible by creating tools for both basic users and those trying to create their own exchanges.
+
+## Debug events
+
+Debug events are a means for seeing what's going on inside of Urql exchanges. Before getting started, here are a few things to be aware of.
+
+### Anyone with client access can listen
+
+This includes the creator of an `urql` client and any of it's exchanges (although the latter is not advised).
+
+### Debug events are disabled in production
+
+Debug events are a fire-and-forget mechanism and should not be used as a means for messaging in your app.
+
+## Consuming debug events
+
+Debugging events can be inspected either graphically using our devtools or manually by subscribing to events on the client.
+
+### Devtools
+
+The quickest way is going to be using our [devtools extension](https://github.com/FormidableLabs/urql-devtools/) which visualizes events on a timeline and provides tools to filter events, inspect the cache, and trigger custom querys via the running client.
+
+![Urql Devtools Timeline](../assets/devtools-timeline.png)
+
+Visit [the repo](https://github.com/FormidableLabs/urql-devtools/) for instructions on getting started.
+
+> Note: Devtools is unfortunately not currently supported for React Native but we're looking into it!
+
+### Manual consumption of events
+
+For those not looking to use a GUI to view events, the client has a `subscribeToDebugTarget` function.
+
+As demonstrated below, the `subscribeToDebugTarget` function takes a callback which is called with every debug event that is dispatched.
+
+```ts
+const { unsubscribe } = client.subscribeToDebugTarget(event => {
+  if (event.source === 'dedupExchange') {
+    return;
+  }
+
+  console.log(event);
+});
+
+// ...
+// Unsubscribe from events
+unsubscribe();
+```
+
+## Dispatching debug events
+
+Debug events are a means of sharing implementation details to consumers of an exchange.
+
+### Dispatching a debug event
+
```

Comment:
We should give this title some content else we should delete it, no?

Reply:
Good spot!